test/transcoding/atomics.spt

```diff
@@ -0,0 +1,95 @@
+3 Source 3 102000 
+3 Name 7 "dst"
+4 Name 8 "object"
+5 Name 9 "expected"
+4 Name 10 "desired"
+3 Decorate 7 Volatile 
+4 TypeInt 3 32 0 
+4 Constant 3 13 1 
+4 Constant 3 14 16 
+4 Constant 3 16 0 
+2 TypeVoid 2 
+4 TypePointer 4 5 3 
+7 TypeFunction 5 2 4 4 4 3 
+2 TypeBool 28 
+
+
+5 Function 2 6 0 5 
+3 FunctionParameter 4 7 
+3 FunctionParameter 4 8 
+3 FunctionParameter 4 9 
+3 FunctionParameter 3 10 
+
+2 Label 11 
+6 AtomicIIncrement 3 12 7 13 14 
+7 AtomicSMax 3 15 7 13 14 16 
+7 AtomicUMax 3 17 7 13 14 16 
+7 AtomicSMin 3 18 7 13 14 16 
+7 AtomicUMin 3 19 7 13 14 16 
+7 AtomicIAdd 3 20 7 13 14 13 
+7 AtomicISub 3 21 7 13 14 13 
+7 AtomicOr 3 22 7 13 14 13 
+7 AtomicXor 3 23 7 13 14 13 
+7 AtomicAnd 3 24 7 13 14 13 
+9 AtomicCompareExchange 3 25 7 13 14 14 13 16 
+7 AtomicExchange 3 26 7 13 14 13 
+6 AtomicLoad 3 27 8 13 14 
+5 AtomicStore 8 13 14 10 
+6 AtomicFlagTestAndSet 28 29 8 13 14 
+4 AtomicFlagClear 8 13 14 
+1 Return 
+
+1 FunctionEnd 
+
+
+; RUN: llvm-spirv %s -to-binary -o %t1.spv
+; RUN: spirv-val %t1.spv
+; RUN: llvm-spirv -r %t1.spv -o %t1.bc -spirv-ocl-builtins-version="CL1.2"
+; RUN: llvm-dis < %t1.bc | FileCheck %s --check-prefix=CHECK-LLVM-12
+
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_incPU3AS1Vi(i32 addrspace(1)* %dst) [[attr:#[0-9]+]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_maxPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_maxPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_minPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_minPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_addPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_subPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z9atomic_orPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_xorPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_andPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z14atomic_cmpxchgPU3AS1Viii(i32 addrspace(1)* %dst, i32 0, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_addPU3AS1Vii(i32 addrspace(1)* %object, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 %desired) [[attr]]
+; CHECK-LLVM-12: %.tmp = call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 1) [[attr]]
+; CHECK-LLVM-12: trunc i32 %.tmp to i1
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 0) [[attr]]
+
+; RUN: llvm-spirv %s -to-binary -o %t2.spv
+; RUN: spirv-val %t2.spv
```

Comment:
It is not necessary to create the same spv file twice and run spirv-val twice for the same file.

Reply:
Sure, thanks.